src/evoKnowledgeShare/evoKnowledgeShare.Backend/Repositories/UserRepository.cs

```diff
@@ -8,53 +10,75 @@ public class UserRepository : Repository<User>
         public UserRepository(EvoKnowledgeDbContext dbContext) : base(dbContext)
         { }
 
-        public override async Task AddAsync(User entity) => await myDbContext.Users.AddAsync(entity);
+        public override async Task<User> AddAsync(User entity)
+        {
+            await myDbContext.Users.AddAsync(entity);
+            await myDbContext.SaveChangesAsync();
+            return myDbContext.Users.FirstOrDefault(x => Equals(x, entity)) ?? throw new NullReferenceException();
+        }
 
-        public override async Task AddRangeAsync(IEnumerable<User> entities) => await myDbContext.Users.AddRangeAsync(entities);
+        public override async Task<IEnumerable<User>> AddRangeAsync(IEnumerable<User> entities)
+        {
+            await myDbContext.Users.AddRangeAsync(entities);
+            await myDbContext.SaveChangesAsync();
+            return myDbContext.Users.Where(user => entities.Any(entity => entity.Id == user.Id));
```

Comment:
NullReferenceException futás időben generálódik, nem figyelmeztetett, hogy így nem kellene használni?
```suggestion
            public override async Task<Topic> AddAsync(Topic entity)
        {
            entity.Id = Guid.NewGuid();
            try
            {
                EntityEntry<Topic> addedTopic = await myDbContext.Topics.AddAsync(entity);
                await myDbContext.SaveChangesAsync();
                return addedTopic.Entity;
            }
            catch (OperationCanceledException ex)
            {
                //TODO: Log(ex)
                throw new OperationCanceledException($"Operation exception happened during function {nameof(AddAsync)}.", ex, CancellationToken.None);
            }
        }
```